### DISCUSSION TITLE
PgSQL extensions

### DISCUSSION THREAD

**Discussion by mrdanielm** [coollabsio/coolify#2226] [Feature Requests]
Add support for PgSQL extensions in the default Postgres, whilst you can use a docker hub image you lose the nice UI that comes with the built in database. Would be good if all the common plugins (Postgis for example) are preinstalled, similar to what [Render](https://docs.render.com/postgresql-extensions)/Heroku etc does.

<img width="1069" alt="image" src="https://github.com/coollabsio/coolify/assets/6180100/11107173-0dd5-4612-9f02-2f39337ff405">

**Comment by ksaitor** [comment]
having a pre-installed `pgvector` and `postgis` would be amazing!

**Comment by HitoshiroKyoka** [comment]
I can put the extensions I want to use in a Docker Image and publish it myself, but it is a hassle. It would be more convenient if it were pre-installed.

**Comment by hwhea** [comment]
+1 to this :)

**Comment by kcc989** [comment]
+1 here

**Comment by spaceworkplatform** [comment]
Please add postgis and pgvector, its basically blocking us right now, because the only way to use them with alpine image is build them and then we lose the built-in usage of coolify controls over db.

**Comment by andrasbacsai** [comment]
I made a small improvement in the next version. 
You can choose predefined images when creating a new Postgresql database.

![image](https://github.com/user-attachments/assets/d6690295-383d-4128-818c-0232c1ff0f1d)

During live-stream, we (chat and me) decided to include these 4. What do you think?

**Reply by mrdanielm** [reply]
That looks great, exactly what I was thinking. Supabase alone would cover most of the extensions required thank you!

**Comment by wgicio** [comment]
Great feature!

**Comment by ksaitor** [comment]
will there be no PostGIS + PGVector at the same time? 
ideally, ofc, would be great for people to be able to install any extension

**Reply by Nedi11** [reply]
+1 to install any extension

**Comment by peaklabs-dev** [comment]
Some nice details: https://x.com/ksaitor/status/1844685693471351076

**Comment by ksaitor** [comment]
any updates here? or is it recommended to use Supabase Postgres with extensions? Any pros/cons of that?

**Comment by wottpal** [comment]
any updates here? is it maybe possible to install extensions in the "initialization scripts" section?

**Comment by ksaitor** [comment]
Solved this issue by using Supabase Postgres image. Comes with all plugins one might need. Works.

**Comment by lucasfcunha** [comment]
+1, there should be an option to add the extensions you want as a pre-install script of the image or something similar.

**Comment by chakrihacker** [comment]
Facing this error with supabase postgres 
> role "supabase_admin" does not exist

**Comment by timothymarois** [comment]
This thread is old. but you can use `pgvector/pgvector:0.8.1-pg18` as the image to have vector working
